src/spec.rs

```diff
@@ -0,0 +1,104 @@
+//! Helper functions defined in the Zcash Protocol Specification.
+
+use std::iter;
+
+use blake2b_simd::{Hash, Params};
+use ff::PrimeField;
+use group::Curve;
+use halo2::{
+    arithmetic::{CurveAffine, CurveExt, FieldExt},
+    pasta::pallas,
+};
+
+use crate::{constants::L_ORCHARD_SCALAR, primitives::sinsemilla};
+
+const PRF_EXPAND_PERSONALIZATION: &[u8; 16] = b"Zcash_ExpandSeed";
+
+/// $\mathsf{ToBase}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod q_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_base(hash: Hash) -> pallas::Base {
+    pallas::Base::from_bytes_wide(hash.as_array())
+}
+
+/// $\mathsf{ToScalar}^\mathsf{Orchard}(x) := LEOS2IP_{\ell_\mathsf{PRFexpand}}(x) (mod r_P)$
+///
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn to_scalar(hash: Hash) -> pallas::Scalar {
+    pallas::Scalar::from_bytes_wide(hash.as_array())
+}
+
+/// Defined in [Zcash Protocol Spec § 4.2.3: Orchard Key Components][orchardkeycomponents].
+///
+/// [orchardkeycomponents]: https://zips.z.cash/protocol/nu5.pdf#orchardkeycomponents
+pub(crate) fn commit_ivk(
+    ak: &pallas::Base,
+    nk: &pallas::Base,
+    rivk: &pallas::Scalar,
+) -> pallas::Scalar {
+    let ivk = sinsemilla::short_commit(
+        "z.cash:Orchard-CommitIvk",
+        iter::empty()
+            .chain(ak.to_le_bits().iter().by_val().take(L_ORCHARD_SCALAR))
```

Comment:
```suggestion
            .chain(ak.to_bytes().iter().flat_map(|b| (0..8).map(|i| (b >> i) & 1 != 0)))
```
Consider using a `bytes_to_bits` helper or the `bitvec` crate.